src/scvi/dataloaders/_data_splitting.py

```diff
@@ -132,6 +165,21 @@ def validate_data_split_with_external_indexing(
     n_train = len(external_indexing[0])
     n_val = len(external_indexing[1])
 
+    if batch_size is not None:
+        num_of_cells = n_train % batch_size
+        if (num_of_cells < 3 and num_of_cells > 0) and not (
+            num_of_cells == 1 and drop_last is True
+        ):
+            warnings.warn(
+                f"Last batch will have a small size of {num_of_cells} "
+                f"samples. Consider changing settings.batch_size or batch_size in model.train "
+                f"from currently {settings.batch_size} to avoid errors during model training "
+                f"or change the given external indices accordingly or use drop_last parameter if "
+                f"there is 1 cell left",
```

Comment:
this one to

Reply:
This is confusing. See comment above - line 179 should be gone.

Reply:
yes